src/CoreShop/Component/Order/Generator/CartPriceRuleVoucherCodeGenerator.php

```diff
@@ -77,17 +77,30 @@ public function generateCodes(CartPriceRuleVoucherGeneratorInterface $generator)
 
     protected static function generateCode(string $letters, int $length): string
     {
-        mt_srand((int)microtime() * 1000000);
-        $i = 0;
-        $code = '';
-
-        while ($i < $length) {
-            $num = mt_rand() % (strlen($letters));
-            $tmp = substr($letters, $num, 1);
-            $code = $code . $tmp;
-            $i++;
+        $code = "";
+        $max = strlen($letters);
+
+        for ($i=0; $i < $length; $i++) {
+            $letter = self::cryptoRandSecure(0, $max-1);
+            $code .= $letters[$letter];
```

Comment:
As we're on PHP 7 here, this can/should be simplified to use the PHP native [`random_int()`](https://www.php.net/manual/de/function.random-int.php) function instead of implementing a custom one (see the update of the [SO answer](https://stackoverflow.com/a/13733588) at the bottom):

```suggestion
            $code .= $letters[random_int(0, $max-1)];
    }
```